### PR TITLE
[proto-build-system Part 2]: Protobuf CI compilation 

### PR DESCRIPTION
**Pull request by Marcial1234**
Part of #247
Unaccounted for in prior build system PRs. 

All PRs that depend on the `third_party` compiled proto will fail CI unless we build the proto before attempting to install project dependencies.

Travis will now install the Golang runtime-library for protos (https://github.com/golang/protobuf) on every build, and a dummy proto will be compiled.

Any errors occurring from the above will block CI (graceful errors was decided against).

Code per:
https://github.com/golang/protobuf
http://google.github.io/proto-lens/installing-protoc.html#linux

### CODE REVIEW

File: .gitignore
Comment (Marcial1234) [inline .gitignore]: compiled go-runtime proto files

**Comment by Marcial1234** [comment]
DO NOT MERGE until change is verified to work

**Comment by Marcial1234** [comment]
Per discussion with @shawnbuso, will attempt to have this change on non-`master` branches and see if Travis picks up these changes

File: third_party/tutorial.proto
Comment (Marcial1234) [inline third_party/tutorial.proto]: TODO: Just have the copyright + package name + syntax declarator

File: third_party/tutorial.proto
Comment (cassierecher) [inline third_party/tutorial.proto]: Why does the file end here?

File: .travis.yml
Comment (cassierecher) [inline .travis.yml]: Is this necessary? If we keep installing it every time, that implies the Travis storage space is temporary.

**Review by cassierecher — CHANGES_REQUESTED** [review]

File: .travis.yml
Comment (Marcial1234) [reply .travis.yml]: From what I understood, Travis spawns a new VM on every commit/PR. Besides setting up the languages (as we do above with 2 version of `Golang`), there's no hard "VM dependencies" that we can have only one time.

read: I do not know how to do it otherwise

File: third_party/tutorial.proto
Comment (Marcial1234) [reply third_party/tutorial.proto]: This will be the base proto for `master` and all future branches. I'll act as a dummy for all CI builds. This is a general solution for the proto dependencies on CI, as otherwise we'd need to have a [branching](https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#conditional-stages) or [graceful errors](https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#conditional-stages) (per PR description).

We will move the finalized proto file once the proto rendering + parsing is stable (and back port it to the other repo) once that code is finalized.

File: .travis.yml
Comment (shawnbuso) [reply .travis.yml]: I think Marcial is correct - when we added bazel builds to Travis we had do re-download and install bazel on every presubmit run like this.

**Comment by Marcial1234** [comment]
Need this PR approved to unblock submitting all others.

**Comment by Marcial1234** [comment]
Now we fully import `tutorial.proto` per #252

**Review by cassierecher — APPROVED** [review]

**Comment by Mike29CZ** [comment]
Fine

File: third_party/tutorial.proto
Comment (Mike29CZ) [reply third_party/tutorial.proto]: 6448185e12ba95c084f364f0063ad480d1451c34

**Review by Mike29CZ — APPROVED** [review]

**Review by Mike29CZ — APPROVED** [review]

**Pull request merged**
